### PR TITLE
add systemd_udev_rules resource (closes #26)

### LINKED ISSUE #26: add systemd_udev_d resource
it'd be nice to be able to configure [udev](http://www.freedesktop.org/software/systemd/man/udev.html) rules under `/etc/udev/rules.d` via a domain-specific resource (i.e. not `cookbook_file` or `template`).

### PR DESCRIPTION
**Pull request by nathwill**
per #26, adds a (fairly naive) resource for managing udev rules.

ref: http://www.freedesktop.org/software/systemd/man/udev.html

### CODE REVIEW

**Comment by nathwill** [comment]
@ranjib thoughts? i'm not super enthused about how much logic is in the rules attribute's callback, but figured that it might as well just be a file or template resource unless it's providing extended validation, and this seemed the clearest way to do it.

File: libraries/provider_systemd_udev_rules.rb [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+        new_resource.updated_by_last_action(f.updated_by_last_action?)
+      end
+    end
+
+    action :disable do
+      r = new_resource
+
+      rule_path = ::File.join(DIR, "#{r.name}.rules")
+      null_path = '/dev/null'
```
Comment (ranjib) [inline libraries/provider_systemd_udev_rules.rb]: if you want :0) 

``` ruby
File::NULL
```

File: libraries/resource_systemd_udev_rules.rb [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    actions :create, :delete, :disable
+    default_action :create
+
+    attribute :rules, kind_of: Array, default: [], callbacks: {
+      'is a valid rule set' => lambda do |specs|
+        specs.is_a?(Array) && specs.all? do |spec|
+          spec.is_a?(Array) && spec.all? do |rule|
+            rule.length == 3 &&
+              rule.keys.all? { |k| %w( key operator value ).include? k } &&
+              %w(
```
Comment (ranjib) [inline libraries/resource_systemd_udev_rules.rb]: may be extract this list of valid keys into a method or constant

File: libraries/resource_systemd_udev_rules.rb [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+                NAME OWNER GROUP MODE SECLABEL RUN LABEL GOTO TAG
+              ).any? { |k| rule['key'].start_with? k } &&
+              %w( == != = += -= := ).include?(rule['operator'])
+          end
+        end
+      end
+    }
+
+    def to_s
+      send(:rules).map do |rules|
```
Comment (ranjib) [inline libraries/resource_systemd_udev_rules.rb]: we cant use `rules` straight?

**Comment by ranjib** [comment]
LGTM

**Pull request merged**